lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -834,20 +834,161 @@ void TypeLoweringVisitor::visitExpr(SubfieldOp op) {
   opsToRemove.push_back(op);
 }
 
-// Gracefully die on subaccess operations
+static IntegerAttr getIntAttr(const APInt &value, MLIRContext *context) {
```

Comment:
Instead of adding this, please change the clients to use the builder for firrtl.constant that takes an  `APSInt`.  It eliminates all the type gymnastics.